### PR TITLE
[SYCL][libdevice] Add simple rand for ease of use in device

### PR DESCRIPTION
**Pull request by jinge90**

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @ErnestoSu 
Could you help review this patch for simple rand in device code?
Thanks very much.

File: libdevice/crt_wrapper.cpp [hunk 1]
```diff
@@ -24,6 +33,59 @@ int memcmp(const void *s1, const void *s2, size_t n) {
   return __devicelib_memcmp(s1, s2, n);
 }
 
+#ifndef __NVPTX__
+
+// This simple rand is for ease of use only, the implementation aligns with
+// LLVM libc rand which is based on xorshift64star pseudo random number
+// generator. If work item number < 1024, each work item has its own internal
```
Comment (ErnestoSu) [inline libdevice/crt_wrapper.cpp:40]: should be `<= 1024`

**Review by ErnestoSu — APPROVED** [review]
LGTM. Thanks!

File: libdevice/crt_wrapper.cpp (hunk 1 above)
Comment (jinge90) [reply libdevice/crt_wrapper.cpp]: Done.
Thanks very much.

**Comment by jinge90** [comment]
The pre-ci failure should not be related to this PR, following error log shows a compfail happens in level zero source:
[2024-04-24T04:03:10.212Z] /netbatch/donb535672_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Build_PR_RHEL/llvm.obj/_deps/unified-runtime-src/source/adapters/level_zero/adapter.cpp:212:20: error: ‘setw’ is not a member of ‘std’

[2024-04-24T04:03:10.212Z]          ss << std::setw(30) << std::right << ZeName;

[2024-04-24T04:03:10.212Z]                     ^~~~

[2024-04-24T04:03:10.212Z] /netbatch/donb535672_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Build_PR_RHEL/llvm.obj/_deps/unified-runtime-src/source/adapters/level_zero/adapter.cpp:212:20: note: 

Hi, @intel/llvm-gatekeepers 
Could you help merge this pr?
Thanks very much.

**Pull request merged**